Fix ClassCastException in FeedItemlistFragment

damoasda opened the pull request:
Fixes a ClassCastException in FeedItemlistFragment if a FeedItemFilter is set and the last feed update has failed.
I just removed some lines of code causing the cast error that don't seem necessary to me.

**Affected versions:** 2.1.1 and develop branch

**Steps to reproduce:**

1. Add a feed
2. Turn off the internet connection (Wifi + Mobile)
3. Go to the subscriptions page and press the feed update button
4. Go to feed item list, you will see the red message "Last Refresh failed"
5. Set a feed item filter, e.g. "Not played"
6. The activity crashes with a ClassCastException, see below. It's not possible to show the feed item list any more unless the feed update was successful again.

**Stack trace:**
```Java
2021-01-13 21:50:39.449 2903-2903/de.danoeh.antennapod.debug E/ItemlistFragment: java.lang.ClassCastException: android.widget.LinearLayout$LayoutParams cannot be cast to android.widget.RelativeLayout$LayoutParams
        at de.danoeh.antennapod.fragment.FeedItemlistFragment.refreshHeaderView(FeedItemlistFragment.java:455)
        at de.danoeh.antennapod.fragment.FeedItemlistFragment.lambda$loadItems$10$FeedItemlistFragment(FeedItemlistFragment.java:547)
        at de.danoeh.antennapod.fragment.-$$Lambda$FeedItemlistFragment$Vp4m3mTpwGXoRmmL126SlzQn6HE.accept(Unknown Source:4)
        at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
        at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:124)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

ByteHamster commented:
Ouch, that's pretty bad. Thanks for fixing. I think it should go to the master branch, though. If we do another bugfix release (not sure if we will), this should be included.

I think it looks a bit strange that the filter message is displayed above the error message. Is there an easy way to change their order? (I think that's what the code was doing before it got broken)

damoasda commented:
> I think it should go to the master branch, though.

Ok, I switched the pull request to the master branch.

> I think it looks a bit strange that the filter message is displayed above the error message. Is there an easy way to change their order? (I think that's what the code was doing before it got broken)

I switched the two view entries in `feeditemlist_header.xml` and it looks good now. I wonder why this wasn't done in the XML file before but manually in the Java code. Did I miss something?

ByteHamster commented:
Thanks for fixing and changing the branch!

> I wonder why this wasn't done in the XML file before but manually in the Java code.

My guess would be that it was previously a RelativeLayout that needed some `android:above`/`android:below` attributes and it was easier to put them in that order.
